src/uct/tcp/tcp_ep.c

```diff
@@ -727,34 +744,36 @@ static ucs_status_t uct_tcp_ep_connect(uct_tcp_ep_t *ep)
 
 ucs_status_t uct_tcp_ep_set_dest_addr(const uct_device_addr_t *dev_addr,
                                       const uct_iface_addr_t *iface_addr,
-                                      struct sockaddr_storage *dest_addr)
+                                      struct sockaddr *dest_addr)
 {
-    uct_tcp_device_addr_t *tcp_dev_addr  = (uct_tcp_device_addr_t*)dev_addr;
-    uct_tcp_iface_addr_t *tcp_iface_addr = (uct_tcp_iface_addr_t*)iface_addr;
-    const struct in_addr loopback_addr   = {
+    uct_tcp_device_addr_t *tcp_dev_addr   = (uct_tcp_device_addr_t*)dev_addr;
+    uct_tcp_iface_addr_t *tcp_iface_addr  = (uct_tcp_iface_addr_t*)iface_addr;
+    const struct in_addr in4addr_loopback = {
         .s_addr = htonl(INADDR_LOOPBACK)
     };
     const void *in_addr;
     ucs_status_t status;
 
-    memset(dest_addr, 0, sizeof(*dest_addr));
-    dest_addr->ss_family = tcp_dev_addr->sa_family;
+    memset(dest_addr, 0,
+           ((tcp_dev_addr->sa_family == AF_INET) ?
+                    sizeof(struct sockaddr_in) :
+                    sizeof(struct sockaddr_in6)));
```

Comment:
can we use a helper function from sock.h to get address len by family?

Reply:
we have helper func:ucs_sockaddr_sizeof which returns size of ipv4/ipv6 address but we don't have helper func to return size for given family.

Reply:
can we call it on tcp_dev_addr?

Reply:
I dont thik so, because tcp_dev_addr is internal structure

Reply:
ok, so maybe add such helper function (af->len) in sock.c?

Reply:
```
+    memset(dest_addr, 0, ((tcp_dev_addr->sa_family == AF_INET) ?
+                          UCS_IPV4_ADDR_LEN : UCS_IPV6_ADDR_LEN));
```
@yosefe  is this simple enough? we used similar other places in the code

Reply:
@yosefe  how about 
```
+    memset(dest_addr, 0, ((tcp_dev_addr->sa_family == AF_INET) ?
+                          UCS_IPV4_SOCKADDR_LEN : UCS_IPV6_SOCKADDR_LEN));
```
will define constants in sock.h

Reply:
in addition, i would add a helper function to get socket address length by address family